### PR TITLE
fix for #8982

### PR DESCRIPTION
**Pull request by GOjayson**
If it is totally wrong please let me know, i'm new and trying to help.

### CODE REVIEW

**Review by lynxplay — CHANGES_REQUESTED** [review]
Welcome to your first paper PR! 

As mini mentioned, this should most likely be behind a configuration option.
We also definitely still want to remove the PublicBukkitValues from the tag before sending.

To add configuration to paper (see minis comment on the issue) you want to be editing https://github.com/PaperMC/Paper/blob/master/patches/server/0005-Paper-config-files.patch. I'd argue we would want boolean toggles for PDC and LootTables separately.

**Comment by GOjayson** [comment]
Okay thank you for the directions!

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
++        // Check if LootTable visibility is enabled and remove LootTable and LootTableSeed tags if necessary
++        boolean hideLootTables = this.level.paperConfig().entities.hideLootTables;
++        if (hideLootTables && tag.contains("LootTable")) {
+             tag.remove("LootTable");
+             tag.remove("LootTableSeed");
+         }
++
++        // Check if PDC functionality is enabled and remove PDC-related tags if necessary
++        boolean hidePdc = this.level.paperConfig().entities.hidePDC;
++        if (hidePdc && tag.contains("PlayerData")) {
```
Comment (Lulu13022002) [inline patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch:72]: This is the wrong tag for the pdc, idk from what this come but the pdc tag is the "PublicBukkitValues" even if the name is not really good.

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
++import org.bukkit.Bukkit;
++import org.bukkit.craftbukkit.CraftServer;
+ import org.slf4j.Logger;
+ 
+ // CraftBukkit start
+@@ -25,6 +30,8 @@ import org.spigotmc.CustomTimingsHandler; // Spigot
+ import co.aikar.timings.MinecraftTimings; // Paper
+ import co.aikar.timings.Timing; // Paper
+ 
++import java.awt.print.Paper;
```
Comment (Lulu13022002) [inline patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch]: Clean your import a little bit to reduce the diff

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+diff --git a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+index 4532f3a0d74feae0a1249b53e1bfbc18a8808b32..bcdcc4b411fc794760589a6e0b8791801daf2c1c 100644
+--- a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
++++ b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+@@ -100,6 +100,10 @@ public class WorldConfiguration extends ConfigurationPart {
+ 
+     public class Entities extends ConfigurationPart {
+         public boolean entitiesTargetWithFollowRange = false;
++
++        public boolean hideLootTables = false;
```
Comment (Lulu13022002) [inline patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch:16]: I don't think this is the right config section. should go below the anticheat.obfuscation.tile-entities/block-entities or somethings like that.

**Comment by GOjayson** [comment]
Hmm, I might have done the patching wrong.

**Review by Lulu13022002 — COMMENTED** [review]
It seems that you have now two patches, when you add a new change (to an already created patch) you need to amend the commit (while the rebase) to avoid creating a duplicate patch but for now you can just drop the first one

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch (hunk 1 above)
Comment (GOjayson) [reply patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch:72]: Ohw wow, yea found it a bit confusing. Thanks.

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch (hunk 2 above)
Comment (GOjayson) [reply patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch]: Will clean up, Thanks

File: patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch (hunk 3 above)
Comment (GOjayson) [reply patches/server/0967-Added-config-to-Hide-loottables-and-PDC.patch:16]: Yea i've just put them somewhere in the config, i'm not farmiliar yet with all config things. Thank you for correcting me, will change it.

**Comment by GOjayson** [comment]
Okay, finallly. I got a headache trying not to push the wrong things. I hope this is better. 

I don't know if the Configuration part is fully right now. I've put it into Anticheat and added the bools to an enity configurationpart.

**Comment by GOjayson** [comment]
Ok should all be fixed now.

**Comment by GOjayson** [comment]
No... nvm. I don't know what happend exactly but my branch is a mess. I'm going to close this PR and open a new one.

**Pull request closed without merging**